applications/audio/broadcast_configuration_tool/src/main.c

```diff
@@ -0,0 +1,504 @@
+		LOG_ERR("Failed to add ext adv data for broadcast source: %d", ret);
+		return ret;
+	}
+
+	ext_adv_buf_cnt += ret;
+
+	/* Add the number of UUIDs */
+	ext_adv_buf[0].data_len = ext_adv_data->uuid_buf->len;
+
+	LOG_DBG("Size of adv data: %d, num_elements: %d", sizeof(struct bt_data) * ext_adv_buf_cnt,
+		ext_adv_buf_cnt);
+
+	*ext_adv_count = ext_adv_buf_cnt;
+
+	return 0;
+}
+
+/*
+ * @brief  The following configures the data for the periodic advertising.
+ *         This includes the Basic Audio Announcement, including the
+ *         BASE [BAP 3.7.2.2] and BIGInfo.
+ *
+ * @param  big_index         Index of the Broadcast Isochronous Group (BIG) to get
+ *                           advertising data for.
+ * @param  pre_adv_data      Pointer to the periodic advertising buffers.
+ * @param  per_adv_buf       Pointer to the bt_data used for periodic advertising.
+ * @param  per_adv_buf_size  Size of @p ext_adv_buf.
+ * @param  per_adv_count     Pointer to the number of elements added to @p adv_buf.
+ *
+ * @return  0 for success, error otherwise.
+ */
+static int per_adv_populate(uint8_t big_index, struct broadcast_source_per_adv_data *pre_adv_data,
+			    struct bt_data *per_adv_buf, size_t per_adv_buf_size,
+			    size_t *per_adv_count)
+{
+	int ret;
+	size_t per_adv_buf_cnt = 0;
+
+	ret = broadcast_source_per_adv_populate(big_index, pre_adv_data, per_adv_buf,
+						per_adv_buf_size - per_adv_buf_cnt);
+	if (ret < 0) {
+		LOG_ERR("Failed to add per adv data for broadcast source: %d", ret);
+		return ret;
+	}
+
+	per_adv_buf_cnt += ret;
+
+	LOG_DBG("Size of per adv data: %d, num_elements: %d",
+		sizeof(struct bt_data) * per_adv_buf_cnt, per_adv_buf_cnt);
+
+	*per_adv_count = per_adv_buf_cnt;
+
+	return 0;
+}
+
+static void audio_send(void const *const data, size_t size, uint8_t num_ch)
+{
+	int ret;
+	static int prev_ret;
+
+	struct le_audio_encoded_audio enc_audio = {.data = data, .size = size, .num_ch = num_ch};
+
+	ret = broadcast_source_send(0, 0, enc_audio);
+
+	if (ret != 0 && ret != prev_ret) {
+		if (ret == -ECANCELED) {
+			LOG_WRN("Sending operation cancelled");
+		} else {
+			LOG_WRN("Problem with sending LE audio data, ret: %d", ret);
+		}
+	}
+
+	prev_ret = ret;
+}
+
+static void broadcast_create(struct broadcast_source_big *broadcast_param,
+			     struct bt_bap_lc3_preset *preset)
+{
+	static enum bt_audio_location location[2] = {BT_AUDIO_LOCATION_FRONT_LEFT,
+						     BT_AUDIO_LOCATION_FRONT_RIGHT};
+	static struct subgroup_config subgroups[2];
+
+	for (size_t i = 0; i < ARRAY_SIZE(subgroups); i++) {
+		subgroups[i].group_lc3_preset = *preset;
+		subgroups[i].num_bises = 2;
+		subgroups[i].context = BT_AUDIO_CONTEXT_TYPE_MEDIA;
+		subgroups[i].location = location;
+	}
+
+	broadcast_param->packing = BT_ISO_PACKING_INTERLEAVED;
+
+	broadcast_param->encryption = false;
+
+	bt_audio_codec_cfg_meta_set_bcast_audio_immediate_rend_flag(
+		&subgroups[0].group_lc3_preset.codec_cfg);
+
+	uint8_t english_src[3] = "eng";
```

Comment:
Be careful, "eng" would be 4 elements including '\0'.
Probably we can replace that to uint8_t english_src[] = "eng"
So we won't trouble by the index issue.

Reply:
sys_get_le24 requires a uint8_t[3] though. But I guess I can change to:
```suggestion
	uint8_t english_src[3] = {'e','n','g'};
```